src/System.Management.Automation/namespaces/LocationGlobber.cs

```diff
@@ -502,7 +502,8 @@ out providerInstance
 
             Collection<string> stringResult = new Collection<string>();
 
-            if (!context.SuppressWildcardExpansion)
+            // if the directory contains wildcard characters, but exists, just return it
+            if (!context.SuppressWildcardExpansion && !Utils.NativeDirectoryExists(userPath))
```

Comment:
Maybe the comment and logic should be "if the directory exists, just return it"?

Reply:
I'm leveraging falling into the `else` clause which is the same as if wildcard expansion is suppressed.  Alternatively, I could rearrange the code so that the suppression/existing happens first which may make it a bit easier to read:

```csharp
if (context.SuppressWildcardExpansion || Utils.NativeDirectoryExists(userPath))
...
```

Reply:
Need to change this code to handle exceptions during check